Sources/PackageGraph/DependencyResolver.swift

```diff
@@ -437,6 +466,14 @@ public struct PackageContainerConstraintSet<C: PackageContainer>: Collection {
         return constraints[identifier] ?? .versionSet(.any)
     }
 
+    public var hashValue: Int {
+        var result = 0
+        for c in self.constraints {
+            result = result &* 31 ^ c.key.hashValue ^ c.value.hashValue
```

Comment:
We should probably multiple the key and value hashValues by another random number before combining them.